src/models/propulsion/FGBrushLessDCMotor.cpp

```diff
@@ -142,42 +145,27 @@ void FGBrushLessDCMotor::Calculate(void)
   }
 
   RPM = Thruster->GetRPM();
-  TorqueRequired = abs(((FGPropeller*)Thruster)->GetTorque()); //units [#*ft]
 
-  CurrentRequired = (TorqueRequired * VelocityConstant) / TorqueConstant;
+  TorqueRequired = abs(((FGPropeller*)Thruster)->GetTorque());           //units [#*ft]
+
 
-  //  total current required must include no load current i0
-  CurrentRequired = CurrentRequired + NoLoadCurrent;
- 
   V = MaxVolts * in.ThrottlePos[EngineNumber];
-  
-  //  Delta RPM = (input voltage - currentRequired * coil resistance) * velocity costant
-  DeltaRPM = (V - CurrentRequired * CoilResistance) * VelocityConstant-RPM;
-
-  //  Torque is MaxTorque (stall torque) at 0 RPM and linearly go to 0 at max RPM (MaxVolts*VelocityCostant)
-  //  MaxTorque = MaxCurrent*torqueconstant/velocityconstant*(1-RPM/maxRPM)
-
-  MaxTorque = MaxCurrent / VelocityConstant * TorqueConstant * (1 - RPM / (MaxVolts* VelocityConstant));
-
-  TorqueAvailable = MaxTorque - TorqueRequired;
-  InertiaTorque = (((DeltaRPM/60)*(2.0 * M_PI))/(max(0.00001, in.TotalDeltaT))) * ((FGPropeller*)Thruster)->GetIxx();
- 
-  //  compute acceleration and deceleration phases:
-  //  Acceleration is due to the max delta torque available and is limited to the inertial forces
-
-  if (DeltaRPM >= 0) {
-    TargetTorque = min(InertiaTorque, TorqueAvailable) + TorqueRequired;
-  } else {
-  //  Deceleration is due to braking force given by the ESC and set by parameter deceleration_time 
-    TargetTorque = TorqueRequired - min(abs(InertiaTorque)/(max(DecelerationFactor,0.01)*30),RPM*TorqueConstant/VelocityConstant/VelocityConstant/CoilResistance);
-  }
 
-  EnginePower = ((2 * M_PI) * RPM * TargetTorque) / 60;   //units [#*ft/s]
-  HP = EnginePower /hptowatts*NMtoftpound;                             // units[HP]
+  double Current = (V - RPM / VelocityConstant) / CoilResistance;        // Equation (4) from Drela's document
+
+  if (Current >= 0)
+    TargetTorque = (Current - NoLoadCurrent) / VelocityConstant * TorqueConstant;    // Assuming Kv == Kq
+  else
+    TargetTorque = (Current + NoLoadCurrent) / VelocityConstant * TorqueConstant; // Assuming Kv == Kq
```

Comment:
I don't see why the sign of `NoLoadCurrent` should be kept the opposite of `Current` ? Its definition is not influenced by the sign of `i` i.e. of `Current`.
![Capture d’écran du 2022-01-30 14-13-09](https://user-images.githubusercontent.com/7502904/151701187-c6e4398e-b726-4880-bbce-bf7a34fd6246.png)

Reply:
No load current is by definition the current necessary for the motor to overcome internal friction : therefore it is always against applied torque  and consequently to current.

Reply:
Anyhow it should be 
`TargetTorque=0;
if(Current>=NoLoadCurrent) =(Current-NoloadCurrent)/VelocityConstant * TorqueConstant;
if(Current<-NoLoadCurrent) =(Current+NoloadCurrent)/VelocityConstant * TorqueConstant;

`
It is correct that when abs(current)<noloadcurrent then torque=0.

Reply:
Or if you prefer I can leave it without checking current direction , the effect is negligeable.......

Reply:
> No load current is by definition the current necessary for the motor to overcome internal friction : therefore it is always against applied torque and consequently to current.

OK. It makes sense but it certainly deserve a comment to explain just what you wrote above.

Reply:
resolved with last 2 commits